### PR TITLE
Add VRMC_node_constraint

### PR DESCRIPTION
**Pull request by aaronfranke**
This PR adds support for the [`VRMC_node_constraint`](https://github.com/vrm-c/vrm-specification/tree/master/specification/VRMC_node_constraint-1.0) glTF extension. This is a part of VRM, but it can also be used outside of VRM. Personally, I am using it for custom avatar constraints outside of VRM.

Both import and export are supported. You can also create constraints from within Godot, but note that the tooling can be a bit awkward because it's designed first and foremost for the importer to create constraints.

This code should support node -> node, bone -> bone, node -> bone, and bone -> node, but I have not thoroughly tested all of these cases. At least I have tested rotation on bone -> bone, and all types of node -> node. It should also have no problem handling constraints between different skeletons.

The code is split into 3 files:

* `vrm_node_constraint_doc_ext.gd`: A GLTFDocumentExtension for import/export.
* `bone_node_constraint.gd`: A constraint class which contains the bulk of the logic and is used both at import/export and at runtime. This is independent of the GLTF importer and can be used standalone if desired.
* `bone_node_constraint_applier.gd`: A very small applier Node class that contains constraints. It acts like an AnimationPlayer does to an Animation. This is independent of the GLTF importer and can be used standalone.

This PR also includes a test scene with aim, roll, and rotation constraints. The cyan cubes are controlled with the animation player, while all other cubes are somehow constrained by the cyan cubes. The animation player rotations seem to not export right from Godot, but otherwise the test scene survives a round-trip conversion:

https://github.com/V-Sekai/godot-vrm/assets/1646875/61f6fc62-1151-45ed-8e79-dbb1136c5729

### CODE REVIEW

File: addons/vrm/node_constraint/bone_node_constraint.gd [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+	if source_node == null or target_node == null:
+		return
+	var source_global_transform: Transform3D = _get_source_global_transform()
+	var target_rest_transform: Transform3D = _get_target_global_rest()
+	var rest_dir: Vector3 = _aim_get_rest_direction(target_rest_transform.basis)
+	var aim_dir: Vector3 = target_rest_transform.origin.direction_to(source_global_transform.origin)
+	if rest_dir.is_zero_approx() or aim_dir.is_zero_approx():
+		return
+	var arc := Quaternion(rest_dir, aim_dir)
+	#var desired_quat = arc * _AIM_AXIS_ORIENTATION_QUATS[aim_or_roll_axis]
```
Comment (fire) [inline addons/vrm/node_constraint/bone_node_constraint.gd:76]: Is this code being used?

File: addons/vrm/node_constraint/bone_node_constraint.gd [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+		printerr("BoneNodeConstraint: Roll axis not set! Must be positive X, Y, or Z.")
+		return
+	# Gather axis-angle information from the source rotation.
+	var source_transform: Transform3D = _get_posed_source_transform()
+	var source_quat: Quaternion = source_transform.basis.get_rotation_quaternion()
+	var source_axis: Vector3 = source_quat.get_axis()
+	var source_angle: float = source_quat.get_angle()
+	# Calculate what we need to apply to the target.
+	var axis_index: int = aim_or_roll_axis - 1 # Vector3.Axis
+	var axis_value = source_axis[axis_index]
```
Comment (fire) [inline addons/vrm/node_constraint/bone_node_constraint.gd:93]: Probably should type hint everything

File: addons/vrm/node_constraint/bone_node_constraint.gd (hunk 1 above)
Comment (aaronfranke) [reply addons/vrm/node_constraint/bone_node_constraint.gd:76]: Oops, dead code. I will remove it.

**Comment by fire** [comment]
@lyuma @aaronfranke Not sure the current status of this VRMC_node_constraint, we merged vrm1 support into master.

**Comment by aaronfranke** [comment]
Looks like these changes were added here https://github.com/V-Sekai/godot-vrm/pull/67

**Pull request closed without merging**